### ISSUE TITLE
get predicted probabilities from already-trained model

### ISSUE THREAD

**Issue by 315386775** [cleanlab/cleanlab#344]
hello. i am new here. i want to konw how to get predicted probabilities from already-trained model.

i have trained the 5 classification pytorch model. if i want to use this for the new dataset (include this 5 classes). how to skip the train or fit stage.  

In other words.  how to use trained model to inference the data.

Looking forward to your reply. 

`for fold, (train_idx, val_idx) in enumerate(kfold.split(train_image_id, train_label)):
    X_train, y_train = train_image_id[train_idx], train_label[train_idx]
    X_val, y_val = train_image_id[val_idx], train_label[val_idx]
    
    tscar_model = TSImgClassifier(''resnet50', 5)
    model = Classifier(tscar_model, params)

    lnl = CleanLearning(clf=model, seed=0)
    # cleanlab finds data and label issues in **any dataset**... in ONE line of code!
    clf = lnl.fit(X_train, y_train)
    val_preds[val_idx, :] = clf.predict_proba(X_val)
    print(accuracy_score(y_val, np.argmax(val_preds[val_idx, :], axis=1)))`

**Comment by cgnorthcutt** [comment]
Hi @315386775,

This question is out of scope for the cleanlab package -- you can use any pre-trained model and we don't support pre-trained models in cleanlab. I can try to point you in the right direction though.

To obtain pre-trained model predicted probabilities, you need to reference the docs for whichever package you're using. Some examples:

* pytorch lightning - https://pytorch-lightning.readthedocs.io/en/latest/advanced/transfer_learning.html
* pytorch - https://pytorch.org/vision/0.8/models.html
* keras - https://towardsdatascience.com/step-by-step-guide-to-using-pretrained-models-in-keras-c9097b647b29

## Specific Example
You can copy my code here. I obtain pred_probs for every pytorch and keras model in < 200 lines of code: https://github.com/cgnorthcutt/benchmarking-keras-pytorch

## If you can't find a pre-trained model and you need to train the model yourself
Then you need to compute the predicted probabilities on the dataset you need to find errors in -- use this tutorial https://docs.cleanlab.ai/v2.0.0/tutorials/pred_probs_cross_val.html

**Comment by cgnorthcutt** [comment]
Hi @315386775 , unfortunately this type of support is unrelated to the cleanlab package so we can't help you, but for learning how to use ML models, I definitely recommend checking out the resources in my previous message.

**Comment by cgnorthcutt** [comment]
Once you have the predicted probabilities, i.e. `pred_probs` and your `labels` you can just follow this tutorial which will run you through a lot of the primary functionality of cleanlab: https://docs.cleanlab.ai/v2.0.0/tutorials/indepth_overview.html
